include/toolchain/xcc.h

```diff
@@ -38,4 +38,140 @@
 
 #define __builtin_unreachable() __ASSERT(0, "Unreachable code")
 
-#endif
+#define __builtin_add_overflow(a, b, output) ({ \
+	_Bool __ret__ = 0; \
+	if (__builtin_types_compatible_p(__typeof__(a), int)) { \
+		__ret__ = __builtin_sadd_overflow(a, b, output); \
+	} else if (__builtin_types_compatible_p(__typeof__(a), \
+						long int)) { \
+		__ret__ = __builtin_saddl_overflow(a, b, output); \
+	} else if (__builtin_types_compatible_p(__typeof__(a), \
+						unsigned int)) { \
+		__ret__ = __builtin_uadd_overflow(a, b, output); \
+	} else if (__builtin_types_compatible_p(__typeof__(a), \
+						unsigned long int)) { \
+		__ret__ = __builtin_uaddl_overflow(a, b, output); \
+	} else { \
+		/* assume overflow for all other types */ \
+		__ret__ = 1; \
```

Comment:
Might be a better idea to have an assertion and a CODE_UNREACHABLE here.  I don't know if a BUILD_ASSERT_MSG() would work here, but if it does, it's always better to fail at compile time rather than at runtime.